zun/container/driver.py

```diff
@@ -99,13 +99,93 @@ def load_capsule_driver():
 class BaseDriver(object):
     """Base class for driver."""
 
-    def __init__(self):
-        self.volume_drivers = {}
-        for driver_name in CONF.volume.driver_list:
-            driver = vol_driver.driver(driver_name)
-            self.volume_drivers[driver_name] = driver
-        self.cpu_allocation_ratio = CONF.compute.cpu_allocation_ratio
-        self.ram_allocation_ratio = CONF.compute.ram_allocation_ratio
+    def update_containers_states(self, context, containers, manager):
```

Comment:
I don't think any of these functions were added as part of my changes, so it's probably a merge artifact. I think after Train, the base driver was refactored to have more of these functions defined in an inherited interface.